client/state/account-recovery/settings/reducer.js

```diff
@@ -77,7 +77,7 @@ const phone = createReducer( null, {
 		convertPhoneResponse( settings.phone ),
 
 	[ ACCOUNT_RECOVERY_SETTINGS_UPDATE_SUCCESS ]: ( state, { target, value } ) =>
-		'phone' === target ? convertPhoneResponse( value ) : state,
+		'phone' === target ? value : state,
```

Comment:
Ah that makes sense. It would be nice to update the endpoint, but I think it's ok as is, maybe with a short comment to explain the mismatch.